OmsAgent/omsagent.py

```diff
@@ -356,13 +359,14 @@ def enable():
 def remove_workspace_configuration():
     """
     This is needed to distinguish between extension removal vs extension upgrade.
-    Its a workaround for waagent upgrade routine calling 'remove' on an old version 
-    before calling 'upgrade' on new extension version issue. 
-    In upgrade case, we need workspace configuration to persist when in 
+    Its a workaround for waagent upgrade routine calling 'remove' on an old version
+    before calling 'upgrade' on new extension version issue.
+    In upgrade case, we need workspace configuration to persist when in
     remove case we need all the files be removed.
 
     This method will remove all the files/folders from the workspace path in Etc and Var.
     """
+    public_settings, _ = get_settings()
```

Comment:
why you need this line?

Reply:
Can you squash all your commits so that they appear as single commit in this PR?

Reply:
Yes. The method needs workspace Id from the public_settings but it's not given a method to access it since it not a global name.